aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -42,6 +42,8 @@ module aptos_framework::object {
     /// The resource is not stored at the specified address.
     const ERESOURCE_DOES_NOT_EXIST: u64 = 7;
 
+    const INIT_GUID_CREATION_NUM: u64 = 0x4000000000000;
+
     /// Maximum nesting from one object to another. That is objects can technically have infinte
     /// nesting, but any checks such as transfer will only be evaluated this deep.
     const MAXIMUM_OBJECT_NESTING: u8 = 8;
```

Comment:
Missed this last time, but this seems to be a weird semantics. I can create arbitrary deep nesting, but then silently from this depth onwards `owns` will return false? This sounds like a source of 'gotcha' errors for users. If there is a limit we should abort at it. But in fact, similar as with the guid limit, I'm not quite understanding why we need to set a limit. Shouldn't gas take care of this?

Reply:
You will hit `EMAXIMUM_NESTING`

I'll adjust the `owns` logic to do the same.

Absolutely arbitrary, but considering that maximum gas may let this transaction run for quite a while, it seems like a pretty reasonable optimization in the framework. If we get feedback that we need more than 8, we can always adjust it.

The motivation for not performing the check manually is that it becomes somewhat expensive to do all the data loads to ensure that a user isn't making a cycle... and making a cycle isn't exactly a violation.

Reply:
I see. I only saw the `owns` logic and the comment is perhaps also misleading. This is a nice example for an invariant. The invariant would be the depth restriction is never violated. Then an abort in `owns` is not necessary / provable unreachable. The spec can be added later.